src/Microsoft.OData.Client/Serialization/HttpWebRequestMessage.cs

```diff
@@ -63,149 +68,144 @@ public class HttpWebRequestMessage : DataServiceClientRequestMessage
+        /// </summary>
         public override string Method
         {
             get
             {
-                return this.effectiveHttpMethod;
+                return _effectiveHttpMethod;
             }
-
             set
             {
                 throw new NotSupportedException();
             }
         }
 
-        /// <summary> Returns the collection of request headers. </summary>
-        public override IEnumerable<KeyValuePair<string, string>> Headers
+        /// <summary>
+        ///  Gets or set the credentials for this request.
+        /// </summary>
+        public override ICredentials Credentials
         {
             get
             {
-                List<KeyValuePair<string, string>> headerValues = new List<KeyValuePair<string, string>>(this.httpRequest.Headers.Count);
-                foreach (string headerName in this.httpRequest.Headers.AllKeys)
-                {
-                    string headerValue = this.httpRequest.Headers[headerName];
-                    headerValues.Add(new KeyValuePair<string, string>(headerName, headerValue));
-                }
-
-                return headerValues;
+                return _handler.Credentials;
             }
-        }
-
-        /// <summary> Returns the underlying HttpWebRequest </summary>
-        public System.Net.HttpWebRequest HttpWebRequest
-        {
-            get
+            set
             {
-                return this.httpRequest;
+                _handler.Credentials = value;
             }
         }
 
-        /// <summary>
-        /// Gets or set the credentials for this request.
-        /// </summary>
-        public override System.Net.ICredentials Credentials
-        {
-            get { return this.httpRequest.Credentials; }
-            set { this.httpRequest.Credentials = value; }
-        }
+
+#if !(NETCOREAPP1_0 || NETCOREAPP2_0)
 
         /// <summary>
         /// Gets or sets the timeout (in seconds) for this request.
         /// </summary>
         public override int Timeout
         {
-            get { return this.httpRequest.Timeout; }
-            set { this.httpRequest.Timeout = (int)Math.Min(Int32.MaxValue, new TimeSpan(0, 0, value).TotalMilliseconds); }
+            get
+            {
+                return (int)_client.Timeout.TotalSeconds;
+            }
+            set
+            {
+                _client.Timeout = new TimeSpan(0, 0, value);
+            }
         }
 
         /// <summary>
-        /// Gets or sets the read and write timeout (in seconds) for this request.
+        /// Gets or sets the read write timeout (in seconds) for this request.
+        /// Setting the value of the read write timeout to be similar to that of timeout value. 
+        /// This property will not be used as it is not available in HttpClient and getting rid of it 
+        /// Will break people who have been using it before. 
         /// </summary>
+        [Obsolete("This method is not found in HttpClient")]

```

Comment:
HttpClient is an internal implementation detail that the users of our library know little or nothing about.
If I had to make reference to such an internal detail, here's how I'd phrase it:

```csharp
/// <summary>
<see cref="HttpWebRequestMessage"/> internally uses <see cref="HttpClient"/> class to 
send HTTP requests and receive responses from a resource identified by the specified URI. 
<see cref="HttpClient"/> class does not support read and write timeout.
Currently, this property just sets the <see cref="Timeout"/> value.
It is retained for backward compatibility and will be dropped in a future major release.
/// </summary>
[System.Obsolete("Use Timeout property instead. Read and write timeout not supported by HttpClient used internally.")]
public override int ReadWriteTimeout
{
    get
    {
        return Timeout;
    }
    set
    {
        Timeout = value;
    }
```